File: src/util/entities/Recipient.ts
/*
	Spacebar: A FOSS re-implementation and extension of the Discord.com backend.
	Copyright (C) 2023 Fosscord and Fosscord Contributors
	
	This program is free software: you can redistribute it and/or modify
	it under the terms of the GNU Affero General Public License as published
	by the Free Software Foundation, either version 3 of the License, or
	(at your option) any later version.
	
	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU Affero General Public License for more details.
	
	You should have received a copy of the GNU Affero General Public License
	along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";

@Entity("recipients")
export class Recipient extends BaseClass {
	@Column()
	@RelationId((recipient: Recipient) => recipient.channel)
	channel_id: string;

	@JoinColumn({ name: "channel_id" })
	@ManyToOne(() => require("./Channel").Channel, {
		onDelete: "CASCADE",
	})
	channel: import("./Channel").Channel;

	@Column()
	@RelationId((recipient: Recipient) => recipient.user)
	user_id: string;

	@JoinColumn({ name: "user_id" })
	@ManyToOne(() => require("./User").User, {
		onDelete: "CASCADE",
	})
	user: import("./User").User;

	@Column({ default: false })
	closed: boolean;

	// TODO: settings/mute/nick/added at/encryption keys/read_state
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Recipient.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("recipients")]
public partial class Recipient
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("channel_id")]
    public long ChannelId { get; set; }

    [Column("user_id")]
    public long UserId { get; set; }

    [Column("closed")]
    public bool Closed { get; set; }

    [ForeignKey("ChannelId")]
    [InverseProperty("Recipients")]
    public virtual Channel Channel { get; set; } = null!;

    [ForeignKey("UserId")]
    [InverseProperty("Recipients")]
    public virtual User User { get; set; } = null!;
}
